net/ip_utils.ts

```diff
@@ -0,0 +1,137 @@
+// Copyright 2018-2025 the Deno authors. MIT license.
+
+const IPV4_REGEX = /^[0-9]{0,3}\.[0-9]{0,3}\.[0-9]{0,3}\.[0-9]{0,3}$/;
+
+export type AddressType = "IPv4" | "IPv6";
+
+function ipToInt(addr: string): number {
+  return addr.split(".").reduce(
+    (acc, octet) => (acc << 8) + parseInt(octet),
+    0,
+  ) >>> 0;
+}
+
+/**
+ * Returns the address type ("IPv4" or "IPv6") for a given remote address string.
+ *
+ * @param remoteAddr - The remote address as a string.
+ * @returns The address type ("IPv4" | "IPv6") or undefined if not recognized.
+ */
+export const distinctRemoteAddr = (
+  remoteAddr: string,
+): AddressType | undefined => {
+  if (isIPv4(remoteAddr)) {
+    return "IPv4";
+  }
+  if (isIPv6(remoteAddr)) {
+    return "IPv6";
+  }
+};
+
+/**
+ * Validates whether a given string is a IPv4 address
+ *
+ * @param addr - IPv4 address in a string format (e.g., "192.168.0.1").
+ * @returns a boolean indicating if the string is a IPv4 address
+ */
+export function isIPv4(addr: string): boolean {
+  return IPV4_REGEX.test(addr);
+}
+
+/**
+ * Validates whether a given string is a IPv6 address
+ *
+ * @param addr - IPv6 address in a string format (e.g., "2001:db8::1").
+ * @returns a boolean indicating if the string is a IPv6 address
+ */
+export function isIPv6(addr: string): boolean {
+  return addr.includes(":");
+}
+
+/**
+ * Validates whether a given string is a valid IPv4 address
+ *
+ * @param addr - IPv4 address in a string format (e.g., "192.168.0.1").
+ * @returns a boolean indicating if the string is a valid IPv4 address
+ */
+export function isValidIPv4(addr: string): boolean {
```

Comment:
Having both `isValidIPv4()` and `isIPv4()` is confusing. We should likely just have one, and I think it should be named `isIPv4()`. Ditto for the IPv6 function.